### ISSUE TITLE
`gherkin/parser` seems to be missing

### ISSUE THREAD

**Issue by majormoses** [cucumber/cucumber-ruby-core#160]
## Summary

It looks like the `gherkin` parser was moved from a standalone gem to something in `cucumber` mono repo but looks like something may not have been updated properly. 
```
/home/travis/.rvm/gems/ruby-2.3.7/gems/cucumber-core-3.2.0/lib/cucumber/core/gherkin/parser.rb:2:in `require': cannot load such file -- gherkin/parser (LoadError)
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/cucumber-core-3.2.0/lib/cucumber/core/gherkin/parser.rb:2:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/cucumber-core-3.2.0/lib/cucumber/core.rb:3:in `require'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/cucumber-core-3.2.0/lib/cucumber/core.rb:3:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/foodcritic-7.1.0/lib/foodcritic.rb:2:in `require'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/foodcritic-7.1.0/lib/foodcritic.rb:2:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/foodcritic-7.1.0/bin/foodcritic:2:in `require_relative'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/foodcritic-7.1.0/bin/foodcritic:2:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-2.3.7/bin/foodcritic:23:in `load'
	from /home/travis/.rvm/gems/ruby-2.3.7/bin/foodcritic:23:in `<main>'
	from /home/travis/.rvm/gems/ruby-2.3.7/bin/ruby_executable_hooks:24:in `eval'
	from /home/travis/.rvm/gems/ruby-2.3.7/bin/ruby_executable_hooks:24:in `<main>'
```

## Expected Behavior

Run without failure

## Current Behavior

~~~
/home/travis/.rvm/gems/ruby-2.3.7/gems/cucumber-core-3.2.0/lib/cucumber/core/gherkin/parser.rb:2:in `require': cannot load such file -- gherkin/parser (LoadError)
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/cucumber-core-3.2.0/lib/cucumber/core/gherkin/parser.rb:2:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/cucumber-core-3.2.0/lib/cucumber/core.rb:3:in `require'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/cucumber-core-3.2.0/lib/cucumber/core.rb:3:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/foodcritic-7.1.0/lib/foodcritic.rb:2:in `require'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/foodcritic-7.1.0/lib/foodcritic.rb:2:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/foodcritic-7.1.0/bin/foodcritic:2:in `require_relative'
	from /home/travis/.rvm/gems/ruby-2.3.7/gems/foodcritic-7.1.0/bin/foodcritic:2:in `<top (required)>'
	from /home/travis/.rvm/gems/ruby-2.3.7/bin/foodcritic:23:in `load'
	from /home/travis/.rvm/gems/ruby-2.3.7/bin/foodcritic:23:in `<main>'
	from /home/travis/.rvm/gems/ruby-2.3.7/bin/ruby_executable_hooks:24:in `eval'
	from /home/travis/.rvm/gems/ruby-2.3.7/bin/ruby_executable_hooks:24:in `<main>'
~~~

It fails to find the library/file.

## Possible Solution

Temp work around add `~> 5.1` to your `Gemfile` or `gemspec` while waiting for a fix.

## Steps to Reproduce (for bugs)

I experienced this issue with running [foodcritic](https://github.com/Foodcritic/foodcritic)

## Your Environment

* Version used: 3.2.0
* Operating System and version: travis (trusty I think)
* Link to your project: unfortunately the project I discovered this in is private but here is a link to foodcritic: https://github.com/Foodcritic/foodcritic

**Comment by majormoses** [comment]
Looks like it was pulled in from here: https://github.com/Foodcritic/foodcritic/blob/v14.1.0/foodcritic.gemspec#L22 will open an issue on that repo

**Comment by majormoses** [comment]
We can probably trace it, I don't have the time right now but maybe a little later this evening I will: https://rubygems.org/gems/gherkin/reverse_dependencies

**Comment by majormoses** [comment]
According to https://rubygems.org/gems/cucumber-core/versions/3.2.0 its pulling in `>= 5.0.0` not `~> 5.1.0`

**Comment by majormoses** [comment]
So the released gem specifies it as :point_up: https://github.com/cucumber/cucumber-ruby-core/blob/v3.2.0/cucumber-core.gemspec#L17 but the master has it pessemistically pinned. I can put together a PR to pin it, wait to have it released, then we should be able to release a v4 that pulls in the new dependency as `~> 6.0` as that is a breaking change.

**Comment by aslakhellesoy** [comment]
Upgrading to v6 is underway btw: https://github.com/cucumber/cucumber-ruby/pull/1313

**Comment by xtrasimplicity** [comment]
Given a few people have experienced this today, and there's likely to be more over the coming days, I've released v3.2.1 (with **just** the pessimistic version change). :)
